src/Authorisation/VaultToken.php

```diff
@@ -32,31 +49,36 @@ class VaultToken
      * @throws tokenException
      *   Throw exception if error is returned
      */
-    public static function getToken($tenantId, $clientId, $clientSecret): Token
+    public function getToken($tenantId, $clientId, $clientSecret): Token
     {
-        $guzzle = new Client();
+        $request = $this->requestFactory->createRequest(
+            'POST',
+            'https://login.microsoftonline.com/'.$tenantId.'/oauth2/token'
+        );
+
+        $params = [
+            'client_id' => $clientId,
+            'client_secret' => $clientSecret,
+            'resource' => 'https://vault.azure.net',
+            'grant_type' => 'client_credentials',
+        ];
+
+        $request->getBody()->write(http_build_query($params, '', '&'));
 
         try {
-            $response = $guzzle->post(
-                'https://login.microsoftonline.com/'.$tenantId.'/oauth2/token',
-                [
-                    'form_params' => [
-                        'client_id' => $clientId,
-                        'client_secret' => $clientSecret,
-                        'resource' => 'https://vault.azure.net',
-                        'grant_type' => 'client_credentials',
-                    ],
-                ]
-            )->getBody()->getContents();
-        } catch (\GuzzleHttp\Exception\ClientException $e) {
-            $error = json_decode($e->getResponse()->getBody()->getContents(), true);
-            throw new TokenException($error['error_description'], $e->getResponse()->getStatusCode(), null, $error['error_uri']);
-        } catch (\GuzzleHttp\Exception\GuzzleException $e) {
+            $response = $this->httpClient->sendRequest($request);
+        } catch (ClientExceptionInterface $e) {
             throw new TokenException($e->getMessage(), $e->getCode());
         }
 
-        $data = json_decode($response, true);
+        $data = json_decode($response->getBody()->getContents(), true);
```

Comment:
Are some validation missing here? Do the body always have json content?. If something is wrong with the json `$data` would be `null`? Which will throw index errors in the code below.

Reply:
Fixed in b9a99b9